javalib/src/main/scala/java/util/Date.scala

```diff
@@ -35,12 +40,44 @@ class Date(var milliseconds: Long)
   def setTime(time: Long): Unit =
     milliseconds = time
 
-  override def toString(): String = s"Date($milliseconds)"
-
   def toInstant(): Instant = Instant.ofEpochMilli(getTime())
+
+  override def toString(): String = {
+    val seconds = milliseconds / 1000L
+    val default = s"Date($milliseconds)"
+    Date.secondsToString(seconds, default)
+  }
 }
 
 object Date {
+  // Provide prerequisite for localtime_r calls.
+  // Applications which must track timezone changes over their lifetime
+  // must do timely subsequent tzset() calls, either directly or through
+  // an occasional localtime().
+
+  tzset()
+
+  private def secondsToString(seconds: Long, default: String): String = Zone {
```

Comment:
```suggestion
  private def secondsToString(seconds: Long, default: => String): String = Zone {
```
Let's make default lazy by using by-name parameter, so we don't need to allocate it if it's not needed

Reply:
Thank you for the improvement.  Every nanosecond counts.